ledger/eval.go

```diff
@@ -701,6 +702,15 @@ func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxnWi
 			// intentionally ignoring error here, fees had to have been enough to get here
 		}
 
+		var ts int64
+		var err error
+		if eval.proto.EnableFirstValidTimeStamp {
+			ts, err = eval.l.BlockTimeStamp(txn.Txn.FirstValid.SubSaturate(1))
+			if err != nil {
```

Comment:
Same thoughts about SubSaturate.

Reply:
And another thought.  I think we don't ever want to error here. Instead, could we put a 0 in the timestamp field, and error on access?  That way, we can handle txns with FirstValid=0, but error if they ask for the timestamp.

Reply:
And, if we're worried about using 0 as a flag value (since I suppose it *could* be the timestamp of the first block), we could error on access by explicitly checking firstValid first, and error if it's zero, not if this value is zero.

Reply:
err on access in `txn FirstValidTime`